python/pyarrow/parquet.py

```diff
@@ -1238,6 +1312,52 @@ def _make_manifest(path_or_paths, fs, pathsep='/', metadata_nthreads=1,
     return pieces, partitions, common_metadata_path, metadata_path
 
 
+class ParquetDatasetV2:
+    """
+    ParquetDataset shim using the Dataset API under the hood.
+    """
+    def __init__(self, path_or_paths, filesystem=None, filters=None,
+                 read_dictionary=None, buffer_size=None):
+        import pyarrow.dataset as ds
+        import pyarrow.fs
+
+        # map old filesystems to new one
+        if isinstance(filesystem, LocalFileSystem):
+            filesystem = pyarrow.fs.LocalFileSystem()
+
+        reader_options = {}
+        if buffer_size:
```

Comment:
```suggestion
        if buffer_size is not None:
```
?

Reply:
In ParquetDataset, `buffer_size` has a default of 0 (which I assume meas the default), so this way this catches both None and 0